### PR TITLE
Update parts 1 and 3 and add a new part 4, covering ES6.

### PR DESCRIPTION
**Pull request by yuri**

### CODE REVIEW

**Comment by yuri** [comment]
@e-schultz @SethDavenport How does this look to you?

File: handout/04-es6.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
+## EcmaScript 6 and TypeScript
+
+The new version of JavaScript, "EcmaScript 6" or "ES6", offers a number of new features that extend the power of the language. (The language we usually call "JavaScript" is actually formally known as "EcmaScript".) ES6 is not widely supported in today's browsers, so it needs to be transpiled to ES5. You can choose between several transpilers, but we'll be using TypeScript, which is what Angular team users to write Angular 2.
```
Comment (e-schultz) [inline handout/04-es6.md]: Angular team users to write Angular 2. -> Angular team uses to write Angular 2.

File: handout/04-es6.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+```ts
+  var log = console.log.bind(console);
+  log('Hello');
+```
+
+You can also achieve the same using `Function.call` and `Function.apply`, but we won't discuss this here.
+
+## Arrow Functions
+
+ES6 does offer us one nice new tool for dealing with `this` madness: arrow functions.
```
Comment (e-schultz) [inline handout/04-es6.md]: I'd be more inclined to word this along the lines of:

'ES6 offers some new syntax for dealing with `this` madness: fat arrow functions'

File: handout/04-es6.md [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+  }
+  submit() {
+    [1, 2].forEach(() => console.log(this.errorMessage));
+  }
+}
+```
+
+Here `this` inside the arrow function refers to the instance variable.
+
+## Inheritance: Just Don't.
```
Comment (e-schultz) [inline handout/04-es6.md]: For a potentially controversial statement, especially if teaching the class to people with a .NET/Java background - the content here feels a little light. 

Is there a concise example that could be worked out. The type of thing that people might default to inheritance to accomplish - and showing another way of accomplishing it?

File: handout/01-introduction.md [hunk 4]
```diff
@@ -1,8 +1,8 @@
-# Part 1: Introduction to AngularJS and ngCourse
+# Part 1: Introduction to AngularJS and ngCourse-Next
 
-Angular is the leading open source JavaScript application framework backed by
-Google. This course ("ngCourse") provides an introduction to AngularJS based
-on our experience at [rangle.io](http://rangle.io).
+AngularJS is the leading open source JavaScript application framework backed by Google. The "1.x" version of AngularJS has been quite widely. The new "Angular 2" version of the framework is currently available as a preview.
+
```
Comment (SethDavenport) [inline handout/01-introduction.md]: s/has been quite widely/has been used quite widely/

(missing 'used')

**Comment by yuri** [comment]
Thanks for the comments. I'll merge it now.

**Pull request merged**